pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java

```diff
@@ -0,0 +1,223 @@
+        long incLong = Long.parseLong(literalContexts.get(2).getLiteral().getStringValue());
+        size = (int)((endLong - startLong) / incLong +1);
+        _longArrayLiteral = new long[size];
+        for (int i = 0; i < size; i++, startLong += incLong) {
+        _longArrayLiteral[i] = startLong;
+        }
+        _intArrayLiteral = null;
+        _floatArrayLiteral = null;
+        _doubleArrayLiteral = null;
+        break;
+      case FLOAT:
+        float startFloat = Float.parseFloat(literalContexts.get(0).getLiteral().getStringValue());
+        float endFloat = Float.parseFloat(literalContexts.get(1).getLiteral().getStringValue());
+        float incFloat = Float.parseFloat(literalContexts.get(2).getLiteral().getStringValue());
+        size = (int) ((endFloat - startFloat) / incFloat + 1);
+        _floatArrayLiteral = new float[size];
+        for (int i = 0 ; i < size; i ++, startFloat += incFloat) {
+          _floatArrayLiteral[i] = startFloat;
+        }
+        _intArrayLiteral = null;
+        _longArrayLiteral = null;
+        _doubleArrayLiteral = null;
+        break;
+      case DOUBLE:
+        double startDouble = Double.parseDouble(literalContexts.get(0).getLiteral().getStringValue());
+        double endDouble = Double.parseDouble(literalContexts.get(1).getLiteral().getStringValue());
+        double incDouble = Double.parseDouble(literalContexts.get(2).getLiteral().getStringValue());
+        size = (int) ((endDouble - startDouble) / incDouble +1);
+        _doubleArrayLiteral = new double[size];
+        for (int i = 0 ; i < size; i ++, startDouble += incDouble) {
+          _doubleArrayLiteral[i] = startDouble;
+      }
+        _intArrayLiteral = null;
+        _longArrayLiteral = null;
+        _floatArrayLiteral = null;
+        break;
+      default:
+        throw new IllegalStateException(
+            "Illegal data type for ArrayGenerateTransformFunction: " + _dataType + ", literal contexts: "
+            + Arrays.toString(literalContexts.toArray()));
+    }
+  }
+
+  public int[] getIntArrayLiteral() {
+    return _intArrayLiteral;
+  }
+
+  public long[] getLongArrayLiteral() {
+    return _longArrayLiteral;
+  }
+
+  public float[] getFloatArrayLiteral() {
+    return _floatArrayLiteral;
+  }
+
+  public double[] getDoubleArrayLiteral() {
+    return _doubleArrayLiteral;
+  }
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return new TransformResultMetadata(_dataType, false, false);
+  }
+
+  @Nullable
+  @Override
+  public Dictionary getDictionary() {
+    return null;
+  }
+
+  @Override
+  public int[] transformToDictIdsSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
```

Comment:
throw new UnsupportedOperationException();

Reply:
will fix it and the relevant functions as well. I was speeding through this yesterday sorry